### ISSUE TITLE
Pass preprocessor defines in newShader

### ISSUE THREAD

**Issue by DUznanski** [love2d/love#1577]
I'd like to be able to load "the same shader" for several different purposes: while large parts of the shader code is the same, there are parts that are unnecessary or counterproductive for certain of the tasks.  Ideally, I'd be able to say something like this:

```lua
local smoother_for_picking = love.graphics.newShader("smoothing.vert", "terrain_picking.frag", {"skip_normals" = true})
```
and have `#define skip_normals true` included at the top of my shader program.

---
A motivating example:

I have a vertex shader that works to create a smoothed version of a heightfield.  For drawing, this requires normals and lighting; to do something like picking, however, I don't need to know normals, just the new positions of the vertices.  Calculating the normals takes only about 10% of the code of the shader, but does involve several matrix multiplications and applications that would not be necessary otherwise.

To prevent duplication of code, I'd like wrap the normal calculations in `#ifdef` to prevent their inclusion where they are unnecessary, but in order to do so I must provide a `#define` - and there isn't a way of doing this currently without loading the shader file and manipulating it myself.

**Comment by TheRealOrangus** [comment]
I believe you can do something like this:
```lua
function loadShader(pathToShader, shaderDefines)
    local prefix = ""
    for name, value in pairs(shaderDefines) do
        prefix = prefix .. "#define " .. name .. " " .. tostring(value) .. "\n"
    end
    local shaderCode = prefix .. love.filesystem.read(pathToShader)
    return love.graphics.newShader(shaderCode)
end

local myShader = loadShader("path/to/my/shader.glsl", {skip_normals = true})
```

**Comment by slime73** [comment]
This is now implemented as of commit 0fa00e0 - with slightly different syntax.
